src/services/output_parsers.py

```
@@ -0,0 +1,35 @@
+        json_pattern = r"\{[^{}]*(?:\{[^{}]*\}[^{}]*)*\}"
+        matches = re.findall(json_pattern, text, flags=re.DOTALL)
+
+        for match in matches:
+            try:
+                # Try to parse each potential JSON object
+                return json.loads(match.strip())
+            except json.JSONDecodeError:
+                continue
+
+        # If no valid JSON found, return empty dict
+        raise ValueError(f"Could not extract valid JSON from text: {text[:200]}...")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)